protocol/custom-gas-token.md

```diff
@@ -0,0 +1,330 @@
+    - Chain Governors are encouraged to audit their implementations.
+    - Developers would need to look at the chain’s docs to understand how to obtain native assets.
+- `OptimismPortal` and `L2ToL1MessagePasser` don’t contain the API anymore to bridge native assets.
+- Native ETH bridging is disabled, now enabled through L1-WETH, minting an `OptimismMintableERC20` representation.
+- Supersim needs to be aware when CGT is being used.
+
+## Alternatives Considered
+
+**Maintain the old design.**
+
+The old design required a token to exist in L1 previously. This also needed to introduce a `depositERC20Transaction` as well as the `GasToken` library that contains the relevant gas token functions to derive in L2. This design restricts native asset management by only minting new assets through a `depositTransaction`, digested by the `op-node` through a system transaction (type 126).
+
+**Modify derivation rules to allow custom minting rules.**
+
+Since tx type 126 only allows native assets mints through a `depositTransaction`, customizing this feature to be flexible enough would require major changes on derivation rules, `op-node`, and execution clients to permit new supply creation. Adding new tx types or new precompiles would cause issues such as divergence in current software, unnecessary breaking of existing tooling, and additional maintenance.
+
+**Modify the execution environment and client software.**
+
+EVM-equivalence means the native asset doesn’t have any custom feature other than the described in the EVM specs. It could be possible for the native asset to be modified to allow features such as blacklisting, whitelisting, staking, and voting at the cost of losing EVM equivalence. This would likely affect the current proof systems and fundamental APIs such as `depositTransaction`. This alternative would case similar issues on software maintenance and existing toolings, as stated above.
+
+**Merge `NativeAssetLiquidity` and `LiquidityController`.**
+
+As an alternative to the proposed architecture, both predeploys might be merged into a single one. 
+
+## Risks & Uncertainties
+
+- Any minters granted to interact with the `LiquidityController` require proper audits to ensure native asset logic can’t be broken in production, as the recovery would be costly.
+    - There are two potential paths to minimize the impact: adding rate limits, and burn the unneeded supply during chain's deployment.
+- There is an open discussion on how to support tokens that have other than 18 decimals. This is a concern for the minter that is placed on top of the `LiquidityController`.
+    - One possible solution might be adding `decimals()` in the `LiquidityController`, or fully handling it through the minters.
```

Comment:
Is solving this a blocker to finalizing/merging this design doc? I think it should be as it feels like a fundamental thing and we want to be certain it doesn't require nontrivial design changes

Reply:
There are examples of chains using tokens that have decimals other than 18 as their gas token: for instance Circle's newly announced chain that will USDC for gas (6 decimals):
https://www.circle.com/blog/introducing-arc-an-open-layer-1-blockchain-purpose-built-for-stablecoin-finance